processing/src/main/java/org/apache/druid/segment/join/JoinFilterAnalyzer.java

```diff
@@ -0,0 +1,743 @@
+        }
+      } else {
+        if (LookupColumnSelectorFactory.VALUE_COLUMN.equals(correlatedColumnNoPrefix)) {
+          correlatedValues = ImmutableList.of(filterValue);
+        } else {
+          correlatedValues = lookupJoinable.getExtractor().unapply(filterValue);
+        }
+      }
+      return correlatedValues;
+    }
+
+    if (clauseForFilteredTable.getJoinable() instanceof IndexedTableJoinable) {
+      IndexedTableJoinable indexedTableJoinable = (IndexedTableJoinable) clauseForFilteredTable.getJoinable();
+      IndexedTable indexedTable = indexedTableJoinable.getTable();
+
+      int filterColumnPosition = indexedTable.allColumns().indexOf(filterColumnNoPrefix);
+      int correlatedColumnPosition = indexedTable.allColumns().indexOf(correlatedColumnNoPrefix);
+
+      if (filterColumnPosition < 0 || correlatedColumnPosition < 0) {
+        return null;
+      }
+
+      if (indexedTable.keyColumns().contains(filterColumnNoPrefix)) {
+        IndexedTable.Index index = indexedTable.columnIndex(filterColumnPosition);
+        IndexedTable.Reader reader = indexedTable.columnReader(correlatedColumnPosition);
+        IntList rowIndex = index.find(filterValue);
+        List<String> correlatedValues = new ArrayList<>();
+        for (int i = 0; i < rowIndex.size(); i++) {
+          int rowNum = rowIndex.getInt(i);
+          correlatedValues.add(reader.read(rowNum).toString());
+        }
+        return correlatedValues;
+      } else {
+        IndexedTable.Reader dimNameReader = indexedTable.columnReader(filterColumnPosition);
+        IndexedTable.Reader correlatedColumnReader = indexedTable.columnReader(correlatedColumnPosition);
+        Set<String> correlatedValueSet = new HashSet<>();
+        for (int i = 0; i < indexedTable.numRows(); i++) {
+          if (filterValue.equals(dimNameReader.read(i).toString())) {
+            correlatedValueSet.add(correlatedColumnReader.read(i).toString());
+          }
+        }
+
+        return new ArrayList<>(correlatedValueSet);
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * For all RHS columns that appear in the join's equiconditions, correlate them with base table columns if possible.
+   *
+   * @param adapter              The adapter for the join. Used to determine if a column is a base table column.
+   * @param tablePrefix          Prefix for a join table
+   * @param clauseForTablePrefix Joinable clause for the prefix
+   * @param equiconditions       Map of equiconditions, keyed by the right hand columns
+   *
+   * @return A list of correlatation analyses for the equicondition RHS columns that reside in the table associated with
+   * the tablePrefix
+   */
+  @Nullable
+  private static List<JoinFilterColumnCorrelationAnalysis> findCorrelatedBaseTableColumns(
+      HashJoinSegmentStorageAdapter adapter,
+      String tablePrefix,
+      JoinableClause clauseForTablePrefix,
+      Map<String, Expr> equiconditions
```

Comment:
equiConditions would match the spelling from JoinConditionAnalysis.

Reply:
Adjusted spelling to equiConditions